pkg/controller/jobframework/reconciler.go

```diff
@@ -346,6 +346,21 @@ func (r *JobReconciler) ReconcileGenericJob(ctx context.Context, req ctrl.Reques
 		}
 	}
 
+	// 4.1 update podSetCount for RayCluster resize
```

Comment:
This is generic code, so it shouldn't have any specifics to RayCluster.

Maybe you can add an interface for SizableJobs with a method `IsDownsizable` that returns true/false.

Reply:
Added